activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -1226,6 +1226,11 @@ def warning_ignored?(warning)
             warning.message.match?(warning_matcher) || warning.code.to_s.match?(warning_matcher)
           end
         end
+
+        def extract_table_ref_from_insert_sql(sql)
+          sql[/into\s("[A-Za-z0-9_."\[\]\s]+"|[A-Za-z0-9_."\[\]]+)\s*/im]
+          $1.strip if $1
+        end
```

Comment:
What do you think about moving this to the `DatabaseStatements` module so that it's in the same place as it's being used?

Reply:
Moved. And this is a good point, I was somewhat mindlessly moving from file-type to file-type from specific adapters to the abstract adapter files. But you are right, this is cleaner to put "next to" its usage